sdks/python/apache_beam/utils/annotations.py

```diff
@@ -90,6 +90,8 @@ def inner(*args, **kwargs):
       message = '%s is %s' % (fnc.__name__, label)
       if label == 'deprecated':
         message += ' since %s' % since
+        if extra_message is not None:
+          message += '. ' + extra_message
       message += '. Use %s instead.'% current if current else '.'
```

Comment:
`extra_message` might be needed even if there is a new version. It could for example point to a documentation page with more information. Adding this bit at the end would make more sense, because the prefix part of the message would have the same structure and users can scan the ending part of the message to get the special instructions.